src/create-transformer.ts

```diff
@@ -1,12 +1,14 @@
-import { computed, onBecomeUnobserved, IComputedValue } from "mobx"
+import { computed, onBecomeUnobserved, IComputedValue, IComputedValueOptions } from "mobx"
 import { invariant, addHiddenProp } from "./utils"
 
 export type ITransformer<A, B> = (object: A) => B
 
-export interface ITransformerParams<A, B> {
-    onCleanup?: (resultObject: B | undefined, sourceObject?: A) => void
-    debugNameGenerator?: (sourceObject?: A) => string
-}
+export type ITransformerParams<A, B> =
+    | {
+          onCleanup?: (resultObject: B | undefined, sourceObject?: A) => void
+          debugNameGenerator?: (sourceObject?: A) => string
+      }
+    | Exclude<IComputedValueOptions<B>, "name">
```

Comment:
Shouldn't this be `Omit`?

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#omittk

Reply:
Thanks for seeing that - I've updated the PR.